barretenberg/cpp/src/barretenberg/ecc/fields/field_conversion.test.cpp

```diff
@@ -0,0 +1,143 @@
+#include "barretenberg/ecc/fields/field_conversion.hpp"
+#include <gtest/gtest.h>
+
+namespace bb::field_conversion_tests {
+
+class FieldConversionTest : public ::testing::Test {
+  public:
+    template <typename T> void check_conversion(T x)
+    {
+        size_t len = bb::field_conversion::calc_num_frs<T>();
+        auto frs = bb::field_conversion::convert_to_bn254_frs(x);
+        EXPECT_EQ(len, frs.size());
+        auto y = bb::field_conversion::convert_from_bn254_frs<T>(frs);
+        EXPECT_EQ(x, y);
+    }
+};
+
+/**
+ * @brief Field conversion test for size_t
+ */
+TEST_F(FieldConversionTest, FieldConversionSizeT)
+{
+    size_t x = 210849;
+    check_conversion(x);
+}
+
+/**
+ * @brief Field conversion test for uint32_t
+ */
+TEST_F(FieldConversionTest, FieldConversionUint32)
+{
+    auto x = static_cast<uint32_t>(1) << 31;
+    check_conversion(x);
+}
+
+/**
+ * @brief Field conversion test for bb::fr
+ */
+TEST_F(FieldConversionTest, FieldConversionFr)
+{
+    bb::fr x1(std::string("9a807b615c4d3e2fa0b1c2d3e4f56789fedcba9876543210abcdef0123456789")); // 256 bits
+    check_conversion(x1);
+
+    bb::fr x2(bb::fr::modulus_minus_two); // modulus - 2
+    check_conversion(x2);
+}
+
+/**
+ * @brief Field conversion test for grumpkin::fr
+ *
+ */
+TEST_F(FieldConversionTest, FieldConversionGrumpkinFr)
+{
+    grumpkin::fr x1(std::string("9a807b615c4d3e2fa0b1c2d3e4f56789fedcba9876543210abcdef0123456789")); // 256 bits
+    check_conversion(x1);
+}
+
+/**
+ * @brief Field conversion test for curve::BN254::AffineElement
+ *
+ */
+TEST_F(FieldConversionTest, FieldConversionBN254AffineElement)
+{
+    curve::BN254::AffineElement x1(1, 2);
+    check_conversion(x1);
+
+    curve::BN254::AffineElement x2(grumpkin::fr::modulus_minus_two, grumpkin::fr::modulus_minus_two);
```

Comment:
I notice you're using `modulus_minus_two` a lot (reasonable) but never `modulus_minus_one` which actually seems like the more natural choice. Any particular reason?

Reply:
modulus_minus_one didn't seem to exist, but modulus_minus_two did. Not sure why that happened in the first place but I could definitely just create modulus minus one.